interfaces.go

```diff
@@ -1,46 +1,96 @@
 package dgraph
 
+type DependencyType string
+
+const (
+	// OrDependency means the dependencies are resolved when all `AND`s and a single `OR` dependency is resolved.
+	// In other words, one OR short-circuits all ORs, but not ANDs.
+	OrDependency DependencyType = "or"
+	// AndDependency means the dependency is required no matter what.
+	AndDependency DependencyType = "and"
```

Comment:
I would replace "no matter what" with "for resolution" or similar.

Reply:
Changed.